test/StringUtils.test.cpp

```diff
@@ -1,8 +1,352 @@
+
+	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<float>(-1));
+	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<float>(-1.0));
+	EXPECT_EQ("-1.100000"s, NAS2D::stringFrom<float>(-1.1f));
+	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<float>(0));
+	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<float>(0.0));
+	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<float>(1));
+	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<float>(1.0));
+	EXPECT_EQ("1.100000"s, NAS2D::stringFrom<float>(1.1f));
+
+	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<double>(-1));
+	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<double>(-1.0));
+	EXPECT_EQ("-1.100000"s, NAS2D::stringFrom<double>(-1.1));
+	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<double>(0));
+	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<double>(0.0));
+	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<double>(1));
+	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<double>(1.0));
+	EXPECT_EQ("1.100000"s, NAS2D::stringFrom<double>(1.1));
+
+	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<long double>(-1));
+	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<long double>(-1.0));
+	EXPECT_EQ("-1.100000"s, NAS2D::stringFrom<long double>(-1.1L));
+	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<long double>(0));
+	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<long double>(0.0));
+	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<long double>(1));
+	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<long double>(1.0));
+	EXPECT_EQ("1.100000"s, NAS2D::stringFrom<long double>(1.1L));
+}
+
+
+TEST(String, stringToFromRoundtrip)
+{
+	using namespace std::literals::string_literals;
+	const std::string bigNum(40, '9');
+
+	EXPECT_EQ(""s, NAS2D::stringFrom<std::string>(NAS2D::stringTo<std::string>("")));
+	EXPECT_EQ("Hello World"s, NAS2D::stringFrom<std::string>(NAS2D::stringTo<std::string>("Hello World")));
+
+	EXPECT_EQ("true"s, NAS2D::stringFrom<bool>(NAS2D::stringTo<bool>("true"s)));
+	EXPECT_EQ("false"s, NAS2D::stringFrom<bool>(NAS2D::stringTo<bool>("false"s)));
+
+	EXPECT_THROW(NAS2D::stringFrom<char>(NAS2D::stringTo<char>(""s)), std::invalid_argument);
+	EXPECT_EQ("A"s, NAS2D::stringFrom<char>(NAS2D::stringTo<char>("A"s)));
+	EXPECT_EQ("B"s, NAS2D::stringFrom<char>(NAS2D::stringTo<char>("B"s)));
+	EXPECT_EQ("0"s, NAS2D::stringFrom<char>(NAS2D::stringTo<char>("0"s)));
+	EXPECT_EQ("1"s, NAS2D::stringFrom<char>(NAS2D::stringTo<char>("1"s)));
+	EXPECT_EQ("2"s, NAS2D::stringFrom<char>(NAS2D::stringTo<char>("2"s)));
+	EXPECT_EQ("3"s, NAS2D::stringFrom<char>(NAS2D::stringTo<char>("3"s)));
+	EXPECT_EQ("4"s, NAS2D::stringFrom<char>(NAS2D::stringTo<char>("4"s)));
+	EXPECT_EQ("5"s, NAS2D::stringFrom<char>(NAS2D::stringTo<char>("5"s)));
+	EXPECT_EQ("6"s, NAS2D::stringFrom<char>(NAS2D::stringTo<char>("6"s)));
+	EXPECT_EQ("7"s, NAS2D::stringFrom<char>(NAS2D::stringTo<char>("7"s)));
+	EXPECT_EQ("8"s, NAS2D::stringFrom<char>(NAS2D::stringTo<char>("8"s)));
+	EXPECT_EQ("9"s, NAS2D::stringFrom<char>(NAS2D::stringTo<char>("9"s)));
+	EXPECT_EQ("\n"s, NAS2D::stringFrom<char>(NAS2D::stringTo<char>("\n"s)));
```

Comment:
This check is also questionable, as `"\n"` may be multiple characters on some platforms.